sdk/identity/azure-identity/azure/identity/_credentials/certificate.py

```diff
@@ -54,9 +56,30 @@ def __init__(self, tenant_id, client_id, certificate_path, **kwargs):
 
         # TODO: msal doesn't formally support passwords (but soon will); the below depends on an implementation detail
         private_key = serialization.load_pem_private_key(pem_bytes, password=password, backend=default_backend())
+        client_credential = {"private_key": private_key, "thumbprint": hexlify(fingerprint).decode("utf-8")}
+        if kwargs.pop("send_certificate", False):
+            try:
+                # the JWT needs the whole chain but load_pem_x509_certificate deserializes only the signing cert
+                chain = extract_cert_chain(pem_bytes)
+                client_credential["public_certificate"] = six.ensure_str(chain)
+            except ValueError as ex:
+                # we shouldn't land here, because load_pem_private_key should have raised when given a malformed file
+                message = 'Found no PEM encoded certificate in "{}"'.format(certificate_path)
+                six.raise_from(ValueError(message), ex)
+
         super(CertificateCredential, self).__init__(
-            client_id=client_id,
-            client_credential={"private_key": private_key, "thumbprint": hexlify(fingerprint).decode("utf-8")},
-            tenant_id=tenant_id,
-            **kwargs
+            client_id=client_id, client_credential=client_credential, tenant_id=tenant_id, **kwargs
         )
+
+
+def extract_cert_chain(pem_bytes):
```

Comment:
public method?

Reply:
Private package.

Reply:
Yes. But if we only want to use the method in this model, shouldn't we use _extract_cert_chain? Or we may want to use it in other models in the future?